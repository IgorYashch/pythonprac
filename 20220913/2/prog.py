array = eval(input())
print(', '.join([str(x) for x in sorted(array)]))
