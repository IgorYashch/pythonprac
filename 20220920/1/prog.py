x = int(input())

checker = ['-'] * 3
if x % 25 == 0:
  if x % 2 == 0:
    checker[0] = '+'
  else:
    checker[1] = '+'
if x % 8 == 0:
  checker[2] = '+'

print("A {} B {} C {}".format(*checker))