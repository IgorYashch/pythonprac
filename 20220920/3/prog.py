n = int(input())

def digits_sum(x):
    s = 0
    while x:
        x, digit = divmod(x, 10)
        s += digit
    return s


k = 0
while k < 3:
    l = 0
    while l < 2:
        mul = (n + k) * (n + l)
        print('{} * {} = {}'.format(n + k, n + l, mul if digits_sum(mul) != 6 else ':=)'), end = ' ')
        l += 1
    else:
        mul = (n + k) * (n + l)
        print('{} * {} = {}'.format(n + k, n + l, mul if digits_sum(mul) != 6 else ':=)'))
    k += 1