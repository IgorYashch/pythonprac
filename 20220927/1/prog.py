print([x for x in range(*eval(input())) if all(x % i for i in range(2, x // 2 + 1)) and x != 1])