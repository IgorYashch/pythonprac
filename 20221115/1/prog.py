def objcount(cls):
    class cls_wrapper(cls):
        counter = 0
        def __init__(self, *args, **kwargs):
            super().__init__(*args, **kwargs)
            self.__class__.counter += 1

        def __del__(self):
            if '__del__' in cls.__dict__:
                super().__del__()
            self.__class__.counter -= 1

    return cls_wrapper

import sys
exec(sys.stdin.read())