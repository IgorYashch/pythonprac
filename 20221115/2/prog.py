class Num:
    def __set__(self, obj, val):
        if 'real' in dir(val):
            obj._data = val
        elif '__len__' in dir(val):
            obj._data = len(val)

    def __get__(self, obj, cls):
        if '_data' in obj.__dict__:
            return obj._data
        else:
            return 0

import sys
exec(sys.stdin.read())