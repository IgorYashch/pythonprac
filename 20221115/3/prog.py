class Alpha:
    __slots__ = [chr(x) for x in range(ord('a'), ord('z') + 1)]

    def __init__(self, **kwargs):
        for attr, val in kwargs.items():
            setattr(self, attr, val)

    def __str__(self):
        result_str = []

        for attr in self.__slots__:
            try:
                val = getattr(self, attr)
            except AttributeError:
                continue
            else:
                result_str.append(f'{attr}: {val}')

        return ', '.join(result_str)

class AlphaQ:
    def __init__(self, **kwargs):
        for attr, val in kwargs.items():
            setattr(self, attr, val)

    def __setattr__(self, name, val):
        if len(name) == 1 and ord(name) >= ord('a') and ord(name) <= ord('z'):
            self.__dict__[name] = val
        else:
            raise AttributeError

    def __str__(self):
        result_str = []

        for attr in (chr(x) for x in range(ord('a'), ord('z') + 1) if chr(x) in self.__dict__):
            result_str.append(f'{attr}: {getattr(self, attr)}')

        return ', '.join(result_str)


import sys
exec(sys.stdin.read())
