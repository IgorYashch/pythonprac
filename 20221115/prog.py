class Alpha:
    __slots__ = [chr(x) for x in range(ord('a'), ord('z') + 1)]

    def __init__(self, **kwargs):
        for attr, val in kwargs.items():
            setattr(self, attr, val)

    def __str__(self):
        result_str = []

        for attr in self.__slots__:
            try:
                val = getattr(self, attr)
            except AttributeError:
                continue
            else:
                result_str.append(f'{attr}: {val}')

        return ', '.join(result_str)


import sys
exec(sys.stdin.read())
