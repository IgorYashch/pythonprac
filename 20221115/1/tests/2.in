from prog import objcount

@objcount
class C:
    k = 0
    def __init__(self):
        self.__class__.k += 1

    def __del__(self):
        self.__class__.k -= 1

clss = [C() for i in range(100)]

print(C.counter == 100 and C.k == 100)

for i in range(42):
    clss.pop()

print(C.counter, C.k)