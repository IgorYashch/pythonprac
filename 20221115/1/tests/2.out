True
58 58
